tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/descartes/profile/descartes_profile.h

```diff
@@ -52,10 +52,10 @@ class DescartesPlanProfile
 
   DescartesPlanProfile() = default;
   virtual ~DescartesPlanProfile() = default;
-  DescartesPlanProfile(const DescartesPlanProfile&) = default;
-  DescartesPlanProfile& operator=(const DescartesPlanProfile&) = default;
-  DescartesPlanProfile(DescartesPlanProfile&&) noexcept = default;
-  DescartesPlanProfile& operator=(DescartesPlanProfile&&) noexcept = default;
+  DescartesPlanProfile(const DescartesPlanProfile&) = delete;
+  DescartesPlanProfile& operator=(const DescartesPlanProfile&) = delete;
+  DescartesPlanProfile(DescartesPlanProfile&&) noexcept = delete;
+  DescartesPlanProfile& operator=(DescartesPlanProfile&&) noexcept = delete;
```

Comment:
Why this change? Should we add a clone?

Reply:
This is to prevent slicing. At the moment we are not making copies so the clone was not needed, but may need to added it in the future.

Reply:
So this is because we are storing the base class in the new planning server? Do you have to do this since you changed them to be Ptrs below? It just seems like copying and modifying them might be a pretty common thing you might want to do. For example if you want to make a couple of profiles that are all the same except for one thing.

Reply:
I could enable them for the implementation classes to allow this but mark them final. What do you think.

Reply:
I think easily copying them is important whether we add a clone or mark them final. I'm a bit confused as to how marking them final helps though.

Reply:
It effects others when inheriting from this class where they cannot overload, but it does not remove the possibility of slicing. I think it is low risk to that an intermediate type would be copied so I will add back the constructors for the implementation classes.

Reply:
Also forgot that having pure virtual methods prevents slicing at the base class but could still occur for intermediate types but that would be on the developer. I will go through and clean up similar classes which a have pure virtual methods in the base class and allow derived classes to be copied.

Reply:
Finally came back across these. I will need to implement custom copy and assignment.

https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c67-a-polymorphic-class-should-suppress-copying

https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c130-for-making-deep-copies-of-polymorphic-classes-prefer-a-virtual-clone-function-instead-of-copy-constructionassignment

Reply:
One exception to deleting the copy and assignment
* Only delete the copy and assignment constructor if the base class does not have any pure virtual methods. If you have a pure virtual set these to default and the compiler handles it.

Reply:
You can now copy derived classes for the profiles.

Reply:
Reverted. The derived classes can be copied.